### PR TITLE
SANS user file parse mask time

### LINKED ISSUE #22338: User file parser needs to parse MASK/TIME/REAR

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
The user file is used to set the masks to be used for a given run. One of the types of mask which is set are time masks which can apply to any of the detector banks in the instrument. Currently two different commands are used in the user file for this MASK/DETECTOR/TIME and MASK/TIME/DETECTOR. The parser only supports the latter of these currently this upgrades it to support both. 

**To test:**
Get the data from here 
[loqdemo.zip](https://github.com/mantidproject/mantid/files/1951964/loqdemo.zip)

- In Mantid open the ISIS SANS 2 interface.
- Load maskfile.txt as the userfile
- Load batch_mode_reduction.csv as the batch file
- On the left goto the settings tab then select the mask tab from across the top.
- Confirm that there is a time mask present on the HAB between 39354.5 41348.5
- Confirm that there is a time mask present on the main detector between 19711.5 21228.5

Fixes #22338 . 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/sans/user_file/user_file_parser.py [hunk 1]
```diff
@@ -894,6 +894,9 @@ def _is_line_mask(self, line):
     def _is_time_mask(self, line):
         return does_pattern_match(self._time_pattern, line)
 
+    def _is_time_mask_2(self, line):
```
Comment (josephframsay) [inline scripts/SANS/sans/user_file/user_file_parser.py]: This function doesn't appear to be used anywhere (and as far as I can tell `_time_pattern_2` doesn't exist - is the plan to use this in future?

**Review by josephframsay — CHANGES_REQUESTED** [review]
These changes are nice, and testing works as expected. Just the one small comment:

**Comment by josephframsay** [comment]
Now looks good to go, nice job

**Review by josephframsay — APPROVED** [review]

**Comment by martyngigg** [comment]
The Windows warning was a known issue with the machine.

**Pull request merged**